### ISSUE TITLE
Query parameters in obj[key] format are not loaded or validated

### ISSUE THREAD

**Issue by Schoonology** [apigee-127/swagger-tools#365]
Because of the way Express and qs handle query parameters, `obj[key]=value` becomes `obj = { key: 'value' }`. If there's no `obj` parameter defined in the Swagger spec, the `obj` value is left undefined. Even once that parameter is defined, `obj[key]` is not validated (trying to enforce `obj[key]` is of type "integer", for example).

Fortunately, however, the UI works _flawlessly_, constructing requests with the proper `obj[key]` syntax for each defined query parameter.

This is required to use swagger-tools with JSON API.

Example spec (complete with including `page` as a hack to at least _load_ the parameters, though neither is validated):

```json
{
    "name": "page",
    "in": "query",
    "required": false,
    "type": "string"
},
{
    "name": "page[number]",
    "description": "Requests the given page number, based on `page[size]`.",
    "in": "query",
    "required": false,
    "type": "integer",
    "minimum": 1
},
{
    "name": "page[size]",
    "description": "Requests this many objects in the response.",
    "in": "query",
    "required": false,
    "type": "integer",
    "minimum": 10
}
```

**Comment by whitlockjc** [comment]
What version of swagger-tools are you using?  We addressed this already, unless there is a bug in the implementation.

**Comment by Schoonology** [comment]
:facepalm: It looks like we weren't as up-to-date as I assumed—we were on 0.9.7. That said, it looks like the results are provided as `page[number]` and `page[size]` verbatim, rather than as an object, `page`. I couldn't find an issue that discussed the changes; was the object version discussed?

Thanks for taking the time to respond to this; sorry for failing to do enough of the due diligence on my end.

**Comment by whitlockjc** [comment]
Not a problem at all.  This was addressed in #294 and was released in `v0.9.11`: https://github.com/apigee-127/swagger-tools/blob/master/RELEASE_NOTES.md#v0911-2016-01-14
